packages/core/src/fhir-to-cda/cda-templates/components/allergies.ts

```diff
@@ -0,0 +1,239 @@
+              formatDateToHumanReadableFormat(allergy.resource.reaction?.[0]?.onset) ??
+              NOT_SPECIFIED,
+          },
+          {
+            "#text":
+              formatDateToHumanReadableFormat(allergy.resource.onsetDateTime) ?? NOT_SPECIFIED,
+          },
+          {
+            "#text": allergy.resource.note?.[0]?.text ?? NOT_SPECIFIED,
+          },
+        ],
+      },
+    },
+  ];
+}
+
+function createEntryFromAllergy(allergy: AugmentedAllergy, referenceId: string): ConcernActEntry {
+  return {
+    act: {
+      _classCode: "ACT",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: allergy.typeOid,
+        extension: extensionValue2014,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: allergy.resource.id,
+      }),
+      code: buildCodeCe({
+        code: "CONC",
+        codeSystem: "2.16.840.1.113883.5.6",
+        displayName: "Concern",
+      }),
+      statusCode: {
+        _code: mapAllergyStatusCode(allergy.resource.clinicalStatus?.coding) ?? "active",
+      },
+      effectiveTime: {
+        low: withoutNullFlavorObject(
+          formatDateToCdaTimestamp(allergy.resource.recordedDate),
+          "_value"
+        ),
+      },
+      entryRelationship: createEntryRelationship(allergy.resource, referenceId),
+    },
+  };
+}
+
+function createEntryRelationship(
+  allergy: AllergyIntolerance,
+  referenceId: string
+): ObservationEntryRelationship {
+  const codeSystem = allergy.code?.coding?.[0]?.system;
+  const systemIsLoinc = isLoinc(codeSystem);
+  return {
+    _typeCode: "SUBJ",
+    observation: {
+      _classCode: "OBS",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: oids.allergyIntoleranceObservation,
+        extension: extensionValue2014,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: allergy.id,
+      }),
+      code: buildCodeCe({
+        code: allergy.code?.coding?.[0]?.code,
+        codeSystem: systemIsLoinc ? loincCodeSystem : codeSystem,
+        codeSystemName: systemIsLoinc ? loincSystemName : undefined,
+        displayName: allergy.code?.coding?.[0]?.display,
+      }),
+      value: buildValueCd(allergy.code, referenceId),
+      entryRelationship: [createReactionEntryRelationship(allergy, referenceId)],
+    },
+  };
+}
+
+function createReactionEntryRelationship(
+  allergy: AllergyIntolerance,
+  referenceId: string
+): ObservationEntryRelationship {
+  const codeSystem = allergy.code?.coding?.[0]?.system;
+  const systemIsLoinc = isLoinc(codeSystem);
+  return {
+    _typeCode: "MFST",
+    _inversionInd: true,
+    observation: {
+      _classCode: "OBS",
+      _moodCode: "EVN",
+      templateId: buildInstanceIdentifier({
+        root: oids.reactionObservation,
+        extension: extensionValue2014,
+      }),
+      id: { _nullFlavor: "NI" },
+      code: buildCodeCe({
```

Comment:
should we be using the manifestation.coding or substance.coding inside the Reaction Entry Relationship?

Reply:
Yea this was related to another comment you left. Using manifestation coding now